marimo/_plugins/ui/_impl/tables/narwhals_table.py

```diff
@@ -212,6 +215,19 @@ def _get_summary_internal(self, column: str) -> ColumnSummary:
                 p75=col.quantile(0.75, interpolation="nearest"),
                 p95=col.quantile(0.95, interpolation="nearest"),
             )
+        if col.dtype == nw.Duration:
+            unit_map = {
+                "ms": col.dt.total_milliseconds,
+                "ns": col.dt.total_nanoseconds,
+                "us": col.dt.total_microseconds,
+            }
+            return ColumnSummary(
+                total=total,
+                nulls=col.null_count(),
+                min=str(unit_map["us"]().min()) + "μs",
+                max=str(unit_map["us"]().max()) + "μs",
+                mean=str(unit_map["us"]().mean()) + "μs",
+            )
```

Comment:
it might be worth factoring out the `total_*` call, so it doesn't get recalculated?
```python
            duration = unit_map["us"]()
            return ColumnSummary(
                total=total,
                nulls=col.null_count(),
                min=str(duration.min()) + "μs",
                max=str(duration.max()) + "μs",
                mean=str(duration.mean()) + "μs",
            )
```